File: Algorithms/Sorting/counting_sort.go
package main

import "fmt"

func counting_sort(seq []int) {
  cnt := [256]int{0}

  for i := 0; i < len(seq); i++ {
    cnt[seq[i]] += 1
  }

  k := 0
  for i := 0; i < 256; i++ {
    for j := cnt[i]; j > 0; j-- {
      seq[k] = i
      k++
    }
  }
}

func main() {
  seq := []int{3, 5, 2, 4, 1}

  counting_sort(seq)
  fmt.Println(seq)
}

File: Algorithms/Sorting/counting_sort.rs
fn counting_sort(seq: &mut [u8; 5]) {
    let mut cnt: [u8; 256] = [0; 256];
    
    for i in 0..seq.len() {
        cnt[seq[i] as usize] += 1;
    }

    let mut k = 0;
    for i in 0..cnt.len() {
        for _j in 0..cnt[i] {
            seq[k] = i as u8;
            k += 1;
        }
    }
}

fn main() {
    let mut seq: [u8; 5] = [3, 5, 2, 4, 1];

    counting_sort(&mut seq);

    for i in &seq {
        print!("{} ", i);
    }
    print!("\n");
}
